### ISSUE TITLE
Error when using GeneralizedDiceLoss

### ISSUE THREAD

**Issue by rmfckaka** [wolny/pytorch-3dunet#6]
`Train Epoch: 93	 [4189/90000	(5%)]	Loss: -0.2955(-0.1699)	 Dice: 0.597926(0.678781)
Train Epoch: 93	 [4229/90000	(5%)]	Loss: -0.0577(-0.1727)	 Dice: 0.694479(0.675190)
Train Epoch: 94	 [4234/90000	(5%)]	Loss: -0.0071(-0.1697)	 Dice: 0.538623(0.673142)
Train Epoch: 94	 [4274/90000	(5%)]	Loss: 0.0896(-0.1638)	 Dice: 0.589383(0.672244)
Train Epoch: 95	 [4279/90000	(5%)]	Loss: -0.3254(-0.1646)	 Dice: 0.803462(0.673217)
Train Epoch: 95	 [4319/90000	(5%)]	Loss: -0.1698(-0.1698)	 Dice: 0.737058(0.676099)
Train Epoch: 96	 [4324/90000	(5%)]	Loss: -0.0295(-0.1686)	 Dice: 0.576744(0.675643)
Train Epoch: 96	 [4364/90000	(5%)]	Loss: -0.2689(-0.1686)	 Dice: 0.784353(0.675700)
Train Epoch: 97	 [4369/90000	(5%)]	Loss: -0.2985(-0.1691)	 Dice: 0.790744(0.676570)
Train Epoch: 97	 [4409/90000	(5%)]	Loss: -0.1008(-0.1714)	 Dice: 0.789671(0.678753)
Train Epoch: 98	 [4414/90000	(5%)]	Loss: -0.4003(-0.1720)	 Dice: 0.770390(0.679049)
Train Epoch: 98	 [4454/90000	(5%)]	Loss: -0.2740(-0.1742)	 Dice: 0.800791(0.681251)
Train Epoch: 99	 [4459/90000	(5%)]	Loss: -0.4628(-0.1756)	 Dice: 0.580660(0.680998)
Train Epoch: 99	 [4499/90000	(5%)]	Loss: 0.2465(-0.1745)	 Dice: 0.536527(0.679128)
`
Will GeneralizedDiceLoss compute loss < 0? I checked it but not found the problem.

**Comment by wolny** [comment]
Hi @rmfckaka. As long as the output from the network is a proper probability distribution, (e.g. 'nn.Sigmoid` in case of binary classification or `nn.Softmax` in case of multi-class classification is applied to the activations of the last convolutional layer) then the `GeneralizedDiceLoss` should never be negative. There is a test in the repo which tests just that by randomly generating `input, target` tensors and computing the `GDL`. See: https://github.com/wolny/pytorch-3dunet/blob/master/tests/test_criterion.py. Most likely there is an error in your training code...
